src/main/java/org/terasology/metalrenegades/world/rivers/RiverToElevationProvider.java

```diff
@@ -30,11 +44,20 @@ public void process(GeneratingRegion region) {
         float[] surfaceHeights = elevation.getInternal();
         float[] riversData = rivers.getInternal();
         float[] humidityData = humidity.getInternal();
+        float[] steepnessData = steepnessNoise.noise(elevation.getWorldArea());
         for (int i = 0; i < surfaceHeights.length; ++i) {
+            float steepness = steepnessData[i];
             float riverFac = TeraMath.clamp(riversData[i]);
-            float riverBedElevation = seaLevel - rivers.maxDepth * (riversData[i] * 4 - 3);
-            surfaceHeights[i] = surfaceHeights[i] * (1 - riverFac) + riverBedElevation * riverFac;
-            humidityData[i] += Math.max(0, 0.1 * (seaLevel - riverBedElevation + 7) * riverFac);
+            float riverBedElevation =
+                    seaLevel - rivers.maxDepth * (riversData[i] * (4 + 4 * steepness) - 3 - 4 * steepness);
+            // Never raise the surface to the river bed, erosion only goes downward
```

Comment:
:+1: valid point, and good explanation why this somehow feels more realistic than before.